tests/pipelines/test_pipelines_common.py

```diff
@@ -59,6 +60,136 @@ def check_same_shape(tensor_list):
     return all(shape == shapes[0] for shape in shapes[1:])
 
 
+class SDFunctionTesterMixin:
```

Comment:
let's also add this texter mixin to below three pipelines so we can also test `UNet3DConditionModel`, `I2VGenXLUNet` and `UNetMotionModel`
* https://github.com/huggingface/diffusers/blob/12004bf3a7d3e77eafe3dd8fad1d458d8e001fab/tests/pipelines/text_to_video_synthesis/test_text_to_video.py
* https://github.com/huggingface/diffusers/blob/12004bf3a7d3e77eafe3dd8fad1d458d8e001fab/tests/pipelines/i2vgen_xl/test_i2vgenxl.py
* https://github.com/huggingface/diffusers/blob/12004bf3a7d3e77eafe3dd8fad1d458d8e001fab/tests/pipelines/animatediff/test_animatediff.py

agree that vae methods can be tested in model tests, feel free to do it here or in a follow-up PR:)

Reply:
Sure